src/gwt/src/org/rstudio/studio/client/workbench/views/output/find/FindInFilesDialog.java

```diff
@@ -374,28 +416,41 @@ public void setState(State dialogState)
       checkboxRegex_.setValue(dialogState.isRegex());
       dirChooser_.setText(dialogState.getPath());
 
-      // indexes refer to corresponding enums, but left as ints for readability
       String includeFilePatterns = StringUtil.join(
             Arrays.asList(dialogState.getFilePatterns()), ", ");
-      if (listPresetFilePatterns_.getValue(0) == includeFilePatterns)
-         listPresetFilePatterns_.setSelectedIndex(0);
-      else if (listPresetFilePatterns_.getValue(1) == includeFilePatterns)
-         listPresetFilePatterns_.setSelectedIndex(1);
-      else if (listPresetFilePatterns_.getValue(2) == includeFilePatterns)
-         listPresetFilePatterns_.setSelectedIndex(2);
+      if (includeFilePatterns == 
+          listPresetFilePatterns_.getValue(Include.AllFiles.ordinal()))
```

Comment:
This is also very verbose. I think all that's needed to simplify this is a method to select the index with the given value. I feel like we already have something like this (and if we don't it should be pretty easy to add!)

Reply:
AFAIK there is no existing method to handle this. Where do you think would be the best place to add this function? `listPresetFilePatterns_` is a gwt ListBox object - we currently have one object that inherits from it -- `FormListBox` that only has one method `void setElementId(id)` - would this be a good place?

Reply:
Yes, I think that'd be a good spot!

Reply:
Alright, ready for review again 👍